lwtime/reltime_test.go

```diff
@@ -0,0 +1,415 @@
+	},
+}
+
+func TestRelTimeParse(t *testing.T) {
+	for _, rtpt := range RelTimeParseTests {
+		t.Run(rtpt.Name, func(t *testing.T) {
+			input := lwtime.RelTime{}
+			err := input.Parse(rtpt.Input)
+			if rtpt.Error == nil {
+				assert.Nil(t, err)
+				assert.Equal(t, rtpt.Output, input)
+				return
+			}
+			assert.Equal(t, rtpt.Error.Error(), err.Error())
+		})
+	}
+}
+
+type RelTimeTime_Test struct {
+	Name                string
+	ReferenceTime       string
+	ReferenceTimeFormat string
+	Input               string
+	Output              string
+	Error               error
+}
+
+var RelTimeTime_Tests = []RelTimeTime_Test{
+	RelTimeTime_Test{
+		"now",
+		"2006-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"now",
+		"2006-02-02T15:04:05-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-10-seconds",
+		"2006-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-10s",
+		"2006-02-02T15:03:55-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"plus-5-minutes",
+		"2006-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"5m",
+		"2006-02-02T15:09:05-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-12-hours-at-hour",
+		"2006-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-12h@h",
+		"2006-02-02T03:00:00-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-1-week-at-week",
+		"2006-01-01T15:04:05-07:00",
+		time.RFC3339,
+		"-1w@w",
+		"2005-12-19T00:00:00-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"plus-1-month-at-day",
+		"2007-01-29T15:04:05-07:00",
+		time.RFC3339,
+		"1MON@d", // case-sensitivity test
+		"2007-03-01T00:00:00-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-2-years",
+		"2009-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-2y",
+		"2007-02-02T15:04:05-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-730-days", // 2 years in days
+		"2009-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-730d",
+		"2007-02-03T15:04:05-07:00",
+		nil,
+	},
+	RelTimeTime_Test{
+		"minus-3000-years",
+		"2009-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-3000y",
+		"",
+		errors.New("unable to construct time object: time predates epoch"),
+	},
+	RelTimeTime_Test{
+		"minus-3-invalid",
+		"2009-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-3x",
+		"",
+		errors.New("unable to construct time object: relative time unit (x) is invalid"),
+	},
+	RelTimeTime_Test{
+		"minus-3-years-at-invalid",
+		"2009-02-02T15:04:05-07:00",
+		time.RFC3339,
+		"-3y@x",
+		"",
+		errors.New("unable to construct time object: snap (x) is not a valid relative time unit"),
+	},
+}
+
+func TestRelTimeTime_(t *testing.T) {
+	for _, rt_t := range RelTimeTime_Tests {
+		t.Run(rt_t.Name, func(t *testing.T) {
+			refTime, err := time.Parse(rt_t.ReferenceTimeFormat, rt_t.ReferenceTime)
+			if err != nil {
+				assert.FailNow(t, "unable to parse reference time")
+			}
+			rt := lwtime.RelTime{}
+			err = rt.Parse(rt_t.Input)
+			// if we're expecting an error don't worry about this
+			if rt_t.Error == nil {
+				assert.Nil(t, err)
+			}
+			outTime, err := rt.Time_(refTime)
```

Comment:
I don't think this is what you really want to do here. I feel that you are trying to define a "Constructor" so that you can transform any given time to a "Relative Time", something like this:
```suggestion
			relTime, err := Relative(refTime)
```

Reply:
Fixed.